odoo/models.py

```diff
@@ -6260,38 +6318,73 @@ def recompute(self, fnames=None, records=None):
             The fields and records to recompute have been determined by method
             :meth:`modified`.
         """
-        def process(field):
-            recs = self.env.records_to_compute(field)
-            if (not recs) or (records is not None and not (records & recs)):
-                return
-            if field.store:
-                # do not force recomputation on new records; those will be
-                # recomputed by accessing the field on the records
-                recs = recs.filtered('id')
-                try:
-                    field.recompute(recs)
-                except MissingError:
-                    existing = recs.exists()
-                    field.recompute(existing)
-                    # mark the field as computed on missing records, otherwise
-                    # they remain forever in the todo list, and lead to an
-                    # infinite loop...
-                    for f in recs.pool.field_computed[field]:
-                        self.env.remove_to_compute(f, recs - existing)
-            else:
-                self.env.cache.invalidate([(field, recs._ids)])
-                self.env.remove_to_compute(field, recs)
+        warnings.warn(
+            "Deprecated method recompute(), use flush_model(), flush_recordset() or env.flush_all() instead",
```

Comment:
Since Odoo 13.0, `recompute()` no longer stores the values of computed fields.
The method `recompute()` became a purely technical artifact in the framework, that's why I made the new methods private.
Developers should really use one of the `flush_X()` methods.